File: Data Structures & Algorithms/validate-parentheses/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let arr =[]
        for (let i = 0; i < s.length; i++){
            if (s[i] == '(' || s[i] == '{' || s[i] == '['){
                arr.push(s[i])
            } else {
                if (s[i] == ')'){
                    if (arr.pop() != '(') return false;
                }
                if (s[i] == '}'){
                    if (arr.pop() != '{') return false;
                }
                if (s[i] == ']'){
                    if (arr.pop() != '[') return false;
                }
            }
        }
        if (arr.length != 0) return false;
        return true;
    }
}

File: Data Structures & Algorithms/clone-graph/submission-1.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val = 0, neighbors = None):
        self.val = val
        self.neighbors = neighbors if neighbors is not None else []
"""

class Solution:
    def cloneGraph(self, node: Optional['Node']) -> Optional['Node']:
        if not node:
            return None
        stack = [node]
        visited = {node:Node(node.val)}
        while stack:
            n = stack.pop()
            for nb in n.neighbors:
                if nb not in visited:
                    visited[nb] = Node(nb.val)
                    stack.append(nb)
                visited[n].neighbors.append(visited[nb])


        return visited[node]
